Enable function-paren-newline rule

Danwakeem opened the pull request:
### Description
Added linter rule for `function-paren-newline`. There are [several types of rules](https://eslint.org/docs/rules/function-paren-newline) `eslint` comes with out of the box. 

I have used `consistent` in the past just because of personal preference but using another rule option could better fit strongloops needs. Let me know what you guys think.

#### Related issues

<!--
Please use the following link syntaxes:

- connect to #49 (to reference issues in the current repository)
- connect to strongloop/loopback#49 (to reference issues in another repository)
-->

- connect to #28
- connect to strongloop/loopback-swagger#127

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

bajtos commented:
Thank you @Danwakeem for the pull request. Based on the style enforced by `prettier` in our new codebase in https://github.com/strongloop/loopback-next, I think the setting value `multiline` makes more sense for old codebase.

_Examples of **correct** code:_

```js
function foo(bar, baz) {}

var foo = function(
  bar,
  baz
) {};

var foo = (bar, baz) => {};

foo(bar, baz, qux);

foo(
  bar,
  baz,
  qux
);

foo(function() {
  return baz;
});
```

_Examples of **incorrect** code:_

```js
function foo(bar,
  baz
) {}

var foo = function(
  bar, baz
) {};

var foo = (
  bar,
  baz) => {};

foo(bar,
  baz);

foo(
  function() {
    return baz;
  }
);
```

@strongloop/sq-lb-apex @raymondfeng thoughts?

bajtos commented:
I tested both `multiline` and `consistent` settings in loopback-datasource-juggler. The `consistent` setting requires less formatting changes to make the code pass `eslint` checks.

In the light of that finding, I am withdrawing my previous comment and agree to go forwards with `consistent` setting.

bajtos commented on eslint.json:
```diff
@@ -70,6 +70,7 @@
     }],
-    "strict": ["error", "global"]
+    "strict": ["error", "global"],
+    "function-paren-newline": ["error", "consistent"]
```
We are trying to keep the list of formatting-related settings sorted, could you please move this new line in between `"func-call-spacing"` and `"indent"` lines?

Danwakeem commented:
@bajtos No problem, I went ahead and just removed the original commit and replaced it with one that has a better message, and more appropriate placement of the `eslint` rule. 

Thanks again for the guidance!
